shared/loadgenerator/loadgenerator.go

```diff
@@ -64,11 +65,19 @@ type GeneratorOptions struct {
 	// LoadFactors defines the multiplier for baseQPS.
 	// Len(loadfactors) defines the number of QPS changes.
 	LoadFactors []float64
+	// FileNamePrefix is the prefix used to identify the stored files in artifacts.
+	// If not empty, this can be used to store cpu/mem profile from loadgenerator.
+	// Typically, we can use t.Name() to differentiate between the tests.
+	FileNamePrefix string
```

Comment:
nit, how about renaming this with something including `cpu/mem/profiler`, so that it's easier to understand

Reply:
I would want to use this for various things. cpu/mem can be one of the few things. Plus, this is just a prefix. We will adding things like -Profile to indicate this is a profile and -Trace for trace. So, I Felt this cam eclose to explainig that. Added a comment to explain that as well.